### DISCUSSION TITLE
How to Learn simNIBS

### DISCUSSION THREAD

**Discussion by Teddy0012** [simnibs/simnibs#347] [2. General]
Hello
I've been using simNIBS for a while and have gone through the tutorials and documentation. I'm particularly interested in the algorithms used in simNIBS, such as charm and electrode registration. Although I'm trying to read the related literature, I find it quite challenging. Could anyone suggest the best way to learn these algorithms by reading the code? Any advice would be greatly appreciated.

**Comment by oulap** [comment]
I think it might be a bit tricky to wrap your head around the segmentation algorithm by reading the code. To get the basic idea of how the segmentation algorithm works you can have a look at the slides here:

[model based segmentation 1](https://oulap.github.io/MIA/lecture_slides/lecture_generative_models_for_segmentation_1/lecture_modelBasedSegmentationI.pdf)

[model based segmentation 2](https://oulap.github.io/MIA/lecture_slides/lecture_generative_models_for_segmentation_2/lecture_modelBasedSegmentationII.pdf)

and p. 31 and forward on this [lecture note](https://oulap.github.io/MIA/book/mia.pdf)

If by electrode registration you are referring to how the electrodes are automatically placed on the scalp of a subject, that is done just by warping the positions from MNI space using a warp that is learned during the segmentation. The lecture note also covers the basics of non-linear registration if you're not familiar with that.

**Reply by Teddy0012** [reply]
Thank you for your help. I will study the materials.
